# We actually should but we don't check for res_send as we know it's not available in uClibc.
# => HAVE_* symbol below get never defined => we use the else-branch
--- src/libs/zbxsysinfo/common/net.c
+++ src/libs/zbxsysinfo/common/net.c
@@ -487,6 +487,7 @@
 		return SYSINFO_RET_FAIL;
 	}
 
+#if defined(HAVE_RES_MKQUERY) && defined(HAVE_RES_SEND)
 	if (-1 == (res = res_mkquery(QUERY, zone, C_IN, type, NULL, 0, NULL, buf, sizeof(buf))))
 	{
 		SET_MSG_RESULT(result, zbx_dsprintf(NULL, "Cannot create DNS query: %s", zbx_strerror(errno)));
@@ -517,6 +518,11 @@
 	_res.retry = retry;
 
 	res = res_send(buf, res, answer.buffer, sizeof(answer.buffer));
+#else
+	/* retrand and retry are ignored */
+	if (-1 == (res = res_query(zone, C_IN, type, answer.buffer, sizeof(answer.buffer))))
+		return SYSINFO_RET_FAIL;
+#endif 
 
 	_res.retrans = saved_retrans;
 	_res.retry = saved_retry;
